File: src/test/resources/class_method_sample.m
@implementation ClassMethodSample
+ (void)classMethodHasDef {
    if (x < y) {
        z = x;
    }
}
+ (void)privateClassMethodHasDef {
    if (x > y) {
        z = y;
    }
    return;
}
+ (double)oneFloatParamOneDoubleParamDoubleRet:(float)p1 dnum:(double)p2 {
    return p2;
}
+ (void)sameNameMethod:(int)p1 {
}
- (void)sameNameMethod:(int)p1 {
}
+ (void)sameNamePrivateMethod:(int)p1 {
}
- (void)sameNamePrivateMethod:(int)p1 {
}
@end

File: src/test/resources/class_method_sample.swift
class ClassMethodSample: NSObject, Protocol1, Protocol2 {

    class func classMethodNoDef() {
    }

    class func classMethodNoDef2(num: Int) -> Int {
    }

    class func classMethodHasDef() {
        if x < y {
            z = x
        }


    }

    class func oneFloatParamOneDoubleParamDoubleRet(p1: Float, dnum p2: Double) -> Double {
        return p2

    }

    class func sameNameMethod(p1: Int) {

    }

    func sameNameMethod(p1: Int) {

    }

    class func privateClassMethodHasDef() {
        if x > y {
            z = y
        }
        return
    }

    class func sameNamePrivateMethod(p1: Int) {

    }

    func sameNamePrivateMethod(p1: Int) {

    }

}
